openml-python-common/src/main/java/com/feedzai/openml/python/modules/SharedModulesParser.java

```diff
@@ -0,0 +1,112 @@
+import java.io.InputStream;
+import java.util.Set;
+
+/**
+ * Class responsible for parsing a XML file in order to retrieve the name of the Python modules to be shared across
+ * sub-interpreters.
+ *
+ * @author Paulo Pereira (paulo.pereira@feedzai.com)
+ * @since 0.1.5
+ */
+public class SharedModulesParser {
+
+    /**
+     * Logger.
+     */
+    private static final Logger logger = LoggerFactory.getLogger(SharedModulesParser.class);
+
+    /**
+     * The name of a XML file with the name of the Python modules to be shared across sub-interpreters.
+     */
+    private final String xmlFile;
+
+    /**
+     * Constructor.
+     *
+     * @param xmlFileName The name of a XML file.
+     */
+    public SharedModulesParser(final String xmlFileName) {
+        this.xmlFile = xmlFileName;
+    }
+
+    /**
+     * Constructor.
+     */
+    public SharedModulesParser() {
+        this("python-packages.xml");
+    }
+
+    /**
+     * Gets the {@link InputStream} of {@link #xmlFile} that exists in the current classpath.
+     *
+     * @return The {@link InputStream} of {@link #xmlFile}.
+     */
+    private InputStream getXMLInputStream() {
+        final ClassLoader classLoader = getClass().getClassLoader();
+        return classLoader.getResourceAsStream(this.xmlFile);
+    }
+
+    /**
+     * Parses the {@link #xmlFile} to retrieve the name of the Python modules to be shared across sub-interpreters.
+     *
+     * @return A {@link Set} with the name of the Python modules to be shared across sub-interpreters.
+     * @throws Exception If there is an error while parsing {@link #xmlFile}.
+     */
+    private Set<String> parseXMLFile() throws Exception {
+        final ImmutableSet.Builder<String> sharedModulesBuilder = ImmutableSet.builder();
+        final DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+        final DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+
+        try (final InputStream xmlFile = getXMLInputStream()) {
+            final Document doc = dBuilder.parse(xmlFile);
+            doc.getDocumentElement().normalize();
+
+            final NodeList nList = doc.getElementsByTagName("package");
+            for (int i = 0; i < nList.getLength(); i++) {
+                sharedModulesBuilder.add(nList.item(i).getFirstChild().getNodeValue());
+            }
+        }
+        return sharedModulesBuilder.build();
+    }
+
+    /**
+     * Retrieves a {@link Set} with the name of the Python modules to be shared across sub-interpreters.
+     *
+     * @return The name of the Python modules to be shared across sub-interpreters.
+     */
+    public Set<String> getSharedModules() {
+        Set<String> sharedModules = ImmutableSet.of();
+        try {
+            sharedModules = parseXMLFile();
+        } catch (final Exception e) {
+            logger.warn("Problem while getting the XML file with the Python modules to be shared.", e);
+        }
+        return sharedModules;
+    }
+}
```

Comment:
Now this was already done, but it was a really bit overkill :)   I'd had used just 1 package per line, or a JSON file. But since it's done, let's go for it

Reply:
I meant something much simpler:
`
my_package_1
my_package_2
`

But it's alright!